csp/csp.go

```diff
@@ -260,14 +260,40 @@ func (sp *StoragePlugin) Serve(ctx context.Context, lis net.Listener) error {
 		sp.server = grpc.NewServer(sp.ServerOpts...)
 
 		// Register the CSI services.
-		if s := sp.Controller; s != nil {
-			csi.RegisterControllerServer(sp.server, s)
-		}
+		// Always register the Identity service
 		if s := sp.Identity; s != nil {
 			csi.RegisterIdentityServer(sp.server, s)
+			log.Debug("Registered Identity Service")
+		}
+
+		// determine which of the controller/node services to register
+		nsvc := true
+		csvc := true
+		if s, ok := gocsi.LookupEnv(ctx, EnvVarNodeSvcOnly); ok {
+			if b, err := strconv.ParseBool(s); err == nil {
+				if !b {
```

Comment:
Hi @codenrhoden, 

Actually, now that I think about it, the logic as it exists is incorrect. You're detecting if `NodeSvcOnly` is false and then setting the `csvc` to false because of that. It should be in reverse. There's a much simpler way to write all this code:

```go
		// Always require the identity service.
		if sp.Identity == nil {
			return errors.New("identity service is required")
		}
		// Either a Controller or Node service should be supplied.
		if sp.Controller == nil && sp.Node == nil {
			return errors.New("either a controller or node service is required")
		}

		// Always register the identity service.
		csi.RegisterIdentityServer(sp.server, sp.Identity)

		// Determine which of the controller/node services to register
		csvc := sp.getEnvBool(ctx, EnvVarCtrlSvcOnly)
		nsvc := sp.getEnvBool(ctx, EnvVarNodeSvcOnly)

		if csvc && nsvc {
			return fmt.Errorf(
				"invalid service restriction: %s=true %s=true",
				EnvVarCtrlSvcOnly, EnvVarCtrlSvcOnly)
		}
		if csvc {
			if sp.Controller == nil {
				return errors.New("controller service is required")
			}
			csi.RegisterIdentityServer(sp.server, sp.Controller)
		} else if nsvc {
			if sp.Node == nil {
				return errors.New("node service is required")
			}
			csi.RegisterIdentityServer(sp.server, sp.Node)
		} else {
			if sp.Controller != nil {
				csi.RegisterIdentityServer(sp.server, sp.Controller)
			}
			if sp.Node != nil {
				csi.RegisterIdentityServer(sp.server, sp.Node)
			}
		}
```

Reply:
You are right that my existing logic should have used `b` instead of `!b`. This is what happen when I try to squeeze something in at the last minute...

I'm happy to take your version - it looks correct. Do you have an objection of me keeping the debug log statements for when a service is registered? I personally liked seeing them in my other plugins, as its easy confirmation (when debug is enabled) of what services have been enabled.

Reply:
Hi @codenrhoden,

I think keeping the statement that a service is disabled makes perfect sense :)

Reply:
Hi @codenrhoden,

Hmm, I'm thinking that the log statements reflecting a service registration should even be at `INFO` level.